aws-android-sdk-mobile-client/src/main/java/com/amazonaws/mobile/client/DummyStore.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2017-2021 Amazon.com, Inc. or its affiliates.
+ * All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.amazonaws.mobile.client;
+
+import java.util.Map;
+
+/**
+ * Default key value store to be replaced during mobile client initialization.
+ */
+final class DummyStore implements KeyValueStore {
```

Comment:
Hm, I'm not understanding.

Previously, we would throw an NPE if the store hadn't been initialized.

Now, we have a non-null store, and it throws a not-initialized exception.

Couldn't we just catch the NPE and wrap it?

Reply:
That means we would need to try-catch all 30 unguarded usage of `mStore` inside `AWSMobileClient`, which I find to be more troublesome than this approach.